src/main/doc/authenticationMechanism.asciidoc

```diff
@@ -572,19 +568,22 @@ public class LoginBacking {
     }
 ----
 
-==== SecurityContext.authenticate Notes ====
+==== SecurityContext.authenticate() Notes ====
+
+Any _LoginToContinue_-annotated _HttpAuthenticationMechanism_, including the two built-in FORM authentication mechanisms, can be triggered via a call to the _SecurityContext.authenticate()_ method. This method is based on the _HttpServletRequest.authenticate()_ method, as defined by [https://jcp.org/aboutJava/communityprocess/final/jsr340/index.html[SERVLET31]], but has been extended to support additional functionality defined by the Servlet Container Profile of [https://jcp.org/aboutJava/communityprocess/mrel/jsr196/index2.html[JASPIC]].
 
-Both a _LoginToContinue_ annotated _HttpAuthenticationMechanism_ and the two build-in FORM authentication mechanisms can be triggered via a call to the _SecurityContext#authenticate()_ method. This is based on the behavior of the _HttpServletRequest#authenticate_ method as defined by the Servlet specification and specifically the Servlet Container Profile of the JASPIC spec.
+The extended behavior is facilitated by the _AuthenticationParameters_ parameter passed to _SecurityContext.authenticate()_. _AuthenticationParameters_ includes a _ewAuthentication_ field.
 
-This specification adds aditional behavior to the _authenticate()_ concept by allowing the call site of that method to provide _AuthenticationParameters_. This includes a _newAuthentication_ parameter, which is described below.
+When _newAuthentication_ is set to _true_, the container MUST discard any state that it holds for a _LoginToContinue_-annotated _HttpAuthenticationMechanism_ and that is associated with the current caller. Specifically, this means that any associated state, such as described for the <<LoginToContinue Annotation>> above, MUST be cleared, and the request must proceed as if processing a new request.
 
-When _newAuthentication_ is set to _true_, the container MUST discard all state it holds for an authentication mechanism that's associated with the current caller. Specifically this means that if the _LoginToContinue_ annotated _HttpAuthenticationMechanism_ or any of the build-in FORM authentication mechanisms has any state such as described for the <<LoginToContinue Annotation>> above, this state MUST be discarded.
+When _newAuthentication_ is set to _false_, the container MUST NOT discard any state that it holds for a _LoginToContinue_-annotated _HttpAuthenticationMechanism_ and that is associated with the current caller. Instead, the container MUST resume the in-progress authentication dialog, based on the  associated state. Specifically, the container MUST:
```

Comment:
HttpAuthentiationMechanism would be better here, although at the moment that's only authentication mechanism we have so the difference is tiny.